pilot/pkg/serviceregistry/kube/controller/ambientindex_test.go

```diff
@@ -787,7 +796,104 @@ func TestRBACConvert(t *testing.T) {
 	}
 }
 
-func addPolicy(t *testing.T, cfg *memory.Controller, name, ns string, selector map[string]string, kind config.GroupVersionKind, modify func(*config.Config)) {
+type ambientTestServer struct {
```

Comment:
This is now common to all the ambient tests. Each test method was doing a bunch of this stuff manually and slightly differently in each place.